kore/test/Test/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -837,6 +839,16 @@ test_andTermsSimplification =
                     (mkElemVar Mock.m)
                 )
             assertEqual "" expected actual
+        , testCase "qq" $ do
+            let expect = undefined
+            actual <-
+                simplifyUnify
+                    (Mock.builtinBool False)
+                    (Mock.inKeysMap (mkElemVar Mock.x) Mock.unitMap)
```

Comment:
I also notice that `Mock.unitMap` isn't fully simplified (it should be `Mock.builtinMap []`) but the unification algorithm expects fully simplified arguments.

```suggestion
                    (Mock.inKeysMap (mkElemVar Mock.x) (Mock.builtinMap []))
```

Reply:
Now `traceM` outputs
```
\and{boolSort{}}(
    /* term: */
    /* Created: Test.Kore.Step.Simplification.AndTerms.termAnd */
    /* Spa */
    \and{boolSort{}}(
        /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{boolSort{}}("false"),
        /* Created: Test.Kore.Step.Simplification.AndTerms.inKeysMap */
        /* Sfa */
        inKeys{}(
            /* Fl Fn D Sfa */ x:testSort{},
            /* Fl Fn D Sfa Cl */ /* builtin: */ unitMap{}()
        )
    ),
\and{boolSort{}}(
    /* predicate: */
    /* D Sfa */ \top{boolSort{}}(),
    /* substitution: */
    \top{boolSort{}}()
))
```

Reply:
> I also notice that `Mock.unitMap` isn't fully simplified (it should be `Mock.builtinMap []`) but the unification algorithm expects fully simplified arguments.

Now `traceM` outputs
```
\and{boolSort{}}(
    /* term: */
    /* Created: Test.Kore.Step.Simplification.AndTerms.termAnd */
    /* Spa */
    \and{boolSort{}}(
        /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{boolSort{}}("false"),
        /* Created: Test.Kore.Step.Simplification.AndTerms.inKeysMap */
        /* Sfa */
        inKeys{}(
            /* Fl Fn D Sfa */ x:testSort{},
            /* Fl Fn D Sfa Cl */ /* builtin: */ unitMap{}()
        )
    ),
\and{boolSort{}}(
    /* predicate: */
    /* D Sfa */ \top{boolSort{}}(),
    /* substitution: */
    \top{boolSort{}}()
))
```